packages/h5wasm/src/H5WasmBufferProvider.tsx

```diff
@@ -13,13 +14,22 @@ interface Props {
   getPlugin?: (name: Plugin) => Promise<ArrayBuffer | undefined>;
 }
 
-function H5WasmProvider(props: PropsWithChildren<Props>) {
+function H5WasmBufferProvider(props: PropsWithChildren<Props>) {
   const { filename, buffer, getExportURL, getPlugin, children } = props;
 
   const prevApiRef = useRef<H5WasmApi>();
 
   const api = useMemo(() => {
-    const newApi = new H5WasmApi(filename, buffer, getExportURL, getPlugin);
+    const remote = getH5WasmRemote();
+    const fileId = remote.openFileBuffer(buffer);
+
+    const newApi = new H5WasmApi(
+      remote,
```

Comment:
Technically, since `remote` is a singleton, I could call `getH5WasmRemote` again inside `H5WasmApi`. However, this signature allows me to pass the worker API object directly when testing (and not call `getH5WasmRemote` at all).